### ISSUE TITLE
Some segments and sections are not mapped in correctly in some MachO files (attached)

### ISSUE THREAD

**Issue by verylazyguy** [Vector35/binaryninja-api#2827]
**Version and Platform (required):**
- Binary Ninja Version: 2.5.3163-dev as well as 2.4.2846
- OS: MacOS
- OS Version: X

**Bug Description:**
Segment/section is not being mapped in from file correctly in MachO file

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Go to 0xfffffe00070054b8
2. See offset = 0xe20000
3. hex(bv.get_segment_at(bv.get_section_by_name('__ctf').start).data_offset)
4. See that `0xea8000 != 0xe20000`

**Expected Behavior:**
The correct data from the file is mapped in.

[kernel.development.t6000.gz](https://github.com/Vector35/binaryninja-api/files/7834079/kernel.development.t6000.gz)

**Comment by plafosse** [comment]
Hmm looks like there overlapping segments in the binary. I'm not sure what the linker does in the case.
```
Name:     __CTF
Cmd:      0x000019
vmaddr:   0xfffffe0007ee8000
vmsize:   0000000000000000
fileoff:  0x00000000e20000
filesize: 0x00000000088000
maxprot:  0x000001
initprot: 0x000001
nsects:   0x000001
flags:    0x000004
```

```
Name:     __LINKEDIT
Cmd:      0x000019
vmaddr:   0xfffffe0007ee8000
vmsize:   0x0000000036a948
fileoff:  0x00000000ea8000
filesize: 0x0000000036a948
maxprot:  0x000001
initprot: 0x000001
nsects:   00000000
flags:    00000000
```
Our algorithm chooses the last one defined

**Comment by plafosse** [comment]
My guess here as the title of the section is "__CTF" is that this is actually a malformed binary.

**Comment by verylazyguy** [comment]
malformed? It's a development kernel downloaded directly from Apple, definitely not malformed. CTF in this case is Compact C Type Format

https://github.com/lovasko/libctf has more information about it

but I guess the CTF segment isn't mapped into the virtual address space by the linker so not sure how to handle that... maybe not show the ctf section at all?

**Comment by plafosse** [comment]
Fixed in 2.5.3193
Fortunately in the Mach-O file format each Segment as a list of corresponding sections. Our fix is to simply omit any section whose corresponding segment isn't mapped into virtual memory. When this happens we do a LogInfo with information about the omitted section.
